### DISCUSSION TITLE
CVE-2020-1717

### DISCUSSION THREAD

**Discussion by ghost** [keycloak/keycloak#28172] [Archive]
Hello,

Our instance of [Dependency-Track](https://dependencytrack.org/) has flagged the keycloak-services 23.0.7 JAR: it says it's impacted by [CVE-2020-1717](https://nvd.nist.gov/vuln/detail/CVE-2020-1717).
There is little info available online about this CVE, so not sure whether it is still valid for recent KC versions.
Maybe there is some info at https://issues.redhat.com/browse/KEYCLOAK-12014, but I cannot access that page.

Do you know about this CVE? Does it still impact KC 23.0.7?

Thanks,
Ben

**Comment by abstractj** [comment]
@canon-cmre-benoit-lecardonnel [CVE-2020-1717](https://nvd.nist.gov/vuln/detail/CVE-2020-1717) is a vulnerability found in Keycloak version 7.0.1, actually we are at Keycloak 24.0.2. We strongly advise our users to upgrade, so they can benefit from the security updates.

**Reply by ghost** [reply]
@abstractj I can't find any definite source of info that indicates the issue is resolved after 7.0.1. Do you confirm that 23+ is not affected?

**Reply by ghost** [reply]
To be 100% clear, this was flagged for the jar in keycloak 23.0.7. We are definitely not using keycloak 7.

**Reply by jonkoops** [reply]
Still upgrade to Keycloak 24 if you want to be secure, we don't support versions of Keycloak other than the current major.

**Reply by canon-cmre-steve-magness** [reply]
So was this vulnerability specifically fixed in Keycloak v24 and versions v7 through v23 are exposed, or was it fixed during/after v7 and v23 is not exposed? Trying to determine if it is a False Positive for v23 given the CVE only mentions v7.0.1 specifically.

I appreciate we should still upgrade to Keycloak 24 eventually, but that's not quite what is being asked.

**Reply by abstractj** [reply]
User enumeration is considered a weakness by our team considering that there are several ways if of mitigating this issue. The CVE mentioned, from what I checked, reported on Keycloak 7 and due to the agreement of handling it as a weakness was not fixed.

For the list of issues around this topic, please check https://github.com/keycloak/keycloak/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+user+enumeration.

**Reply by ghost** [reply]
> there are several ways if of mitigating this issue. 

Can you explain what the mitigations are for this issue? I'd like to check we have them in place in our installations.

Thanks

**Reply by codespearhead** [reply]
A more straightforward report of this CVE can be found [here](https://github.com/advisories/GHSA-rvfc-g8j5-9ccf).

As explained in #17629 , [this type of brute force attack in particular will only affect Keycloak if there's no WAF in front of it doing its part](https://github.com/keycloak/keycloak/issues/17629#issuecomment-1998988468).

**Reply by renannprado** [reply]
We're currently using Keycloak 22 (soon upgrading to 26) and I can confirm this is still an issue and we're looking into how to mitigate the issue.

edit:

At the moment I'm considering creating a custom required action and re-implementing the email change flow.

**Comment by brokeninternet** [comment]
Still an issue in 25.0.01, 
It comes up on every vulnerability scanner as an issue.

Login attempt with wrong email -> 50ms to 60ms answer time.
Login attempt with correct email and wrong password -> 200ms to 300ms answer time.

This flaw combinend with [CVE-2024-4629](https://www.cve.org/CVERecord?id=CVE-2024-4629) is more than an issue.

i crawled on every piece of information about [CVE-2020-1717](https://www.cve.org/CVERecord?id=CVE-2020-1717) and if its fixed already in the latest 26 Version and i cannot find any answer about it. Not at Readhead nor Github or NIST.

Does somebody have any clue what the actual state is? 

EDIT:
should be "fixed" in 25.0.01 but it isn't. - https://github.com/keycloak/keycloak/issues/12298
